src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -129,6 +128,17 @@ function BaseValidateCodeForm(props) {
         input2FARef.current.clear();
     }, [twoFactorAuthCode]);
 
+    useEffect(() => {
```

Comment:
This will re-render the component 30 times / every second as long as a timer's running. Why not instead just use a `setTimeout` with `30000` ms of delay instead of `1000`?